### PR TITLE
[CPS-3082] fix(status): expose event issue when replicas of daemonset is zero

### PR DESCRIPTION
**Pull request by whalecold**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by whalecold** [comment]
@zxq-bit @hezhizhen 
/cherrypick release-1.14

File: status/apps/v1/longrunning.go [hunk 1]
```diff
@@ -76,6 +76,19 @@ func (d *longRunning) Judge() (resStatus releaseapi.ResourceStatus, retErr error
 	// judge status from pods
 	judgeFromPods := d.judge(d.delegate.DesiredReplics(), updated, old)
 
+	// If replicas of daemonset is zero, should check event issue and revision issue
+	// bacause of daemonset's replicas should not be zero.
```
Comment (hezhizhen) [inline status/apps/v1/longrunning.go]: 去掉 because of 的 of

File: status/apps/v1/longrunning.go [hunk 2]
```diff
@@ -76,6 +76,19 @@ func (d *longRunning) Judge() (resStatus releaseapi.ResourceStatus, retErr error
 	// judge status from pods
 	judgeFromPods := d.judge(d.delegate.DesiredReplics(), updated, old)
 
+	// If replicas of daemonset is zero, should check event issue and revision issue
```
Comment (hezhizhen) [inline status/apps/v1/longrunning.go]: 去掉 should

File: status/apps/v1/longrunning.go [hunk 3]
```diff
@@ -76,6 +76,19 @@ func (d *longRunning) Judge() (resStatus releaseapi.ResourceStatus, retErr error
...
+	// bacause of daemonset's replicas should not be zero.
+	if lastEvent != nil && lastEvent.InvolvedObject.Kind == "DaemonSet" &&
+		judgeFromPods.Phase != releaseapi.ResourceRunning &&
+		len(updated)+len(old) == 0 {
```
Comment (hezhizhen) [inline status/apps/v1/longrunning.go]: 调整一下顺序，把 replicas为0的判断放到最前面

File: status/apps/v1/longrunning.go [hunk 4]
```diff
@@ -76,6 +76,19 @@ func (d *longRunning) Judge() (resStatus releaseapi.ResourceStatus, retErr error
...
+		if predictEventsIssue != nil {
```
Comment (hezhizhen) [inline status/apps/v1/longrunning.go:83]: 这两谁先谁后有影响吗？

**Review by hezhizhen — APPROVED** [review]

File: status/apps/v1/longrunning.go (hunk 4 above)
Comment (whalecold) [reply status/apps/v1/longrunning.go:83]: 没有影响

**Comment by whalecold** [comment]
已改 @hezhizhen

**Pull request merged**